File: MyClub/MyClub.Model/Responses/DashboardMostSoldProductResponse.cs
using System;

namespace MyClub.Model.Responses
{
    public class DashboardMostSoldProductResponse
    {
        public int ProductId { get; set; }
        public string ProductName { get; set; }
        public string Category { get; set; }
        public int TotalSold { get; set; }
        public decimal TotalRevenue { get; set; }
        public string ImageUrl { get; set; }
    }
} 

File: MyClub/UI/myclub_desktop/lib/models/dashboard/dashboard_most_sold_product_response.dart
class DashboardMostSoldProductResponse {
  final int productId;
  final String productName;
  final String category;
  final int totalSold;
  final double totalRevenue;
  final String? imageUrl;

  DashboardMostSoldProductResponse({
    required this.productId,
    required this.productName,
    required this.category,
    required this.totalSold,
    required this.totalRevenue,
    this.imageUrl,
  });

  factory DashboardMostSoldProductResponse.fromJson(Map<String, dynamic> json) {
    return DashboardMostSoldProductResponse(
      productId: json['productId'],
      productName: json['productName'],
      category: json['category'],
      totalSold: json['totalSold'],
      totalRevenue: json['totalRevenue']?.toDouble() ?? 0.0,
      imageUrl: json['imageUrl'],
    );
  }
}
